app/components/ImageCropper/ImageCropper.tsx

```diff
@@ -0,0 +1,360 @@
+    setCrop(centerAspectCrop(width, height, givenAspectOrFull));
+  }
+
+  // https://stackoverflow.com/questions/18761404/how-to-scale-images-on-a-html5-canvas-with-better-interpolation
+
+  useDebounceEffect(
+    async () => {
+      if (
+        completedCrop?.width &&
+        completedCrop?.height &&
+        imgRef.current &&
+        previewCanvasRef.current
+      ) {
+        canvasPreview(
+          imgRef.current,
+          previewCanvasRef.current,
+          completedCrop,
+          scale,
+          0
+        );
+      }
+    },
+    100,
+    [completedCrop, scale]
+  );
+
+  async function scaleDown(canvas: HTMLCanvasElement, width: number) {
+    const targetCanvas = document.createElement("canvas") as HTMLCanvasElement;
+    const canvasAspect = canvas.width / canvas.height;
+    const isLandScape = canvas.width > canvas.height;
+    targetCanvas.width = Math.ceil(width * (isLandScape ? 1 : canvasAspect));
+    targetCanvas.height = Math.ceil(width / (!isLandScape ? 1 : canvasAspect));
+
+    const pica = new Pica();
+    return await pica.resize(canvas, targetCanvas);
+  }
+
+  async function handleSave(e: React.SyntheticEvent<HTMLButtonElement>) {
+    e.preventDefault();
+    let canvas = previewCanvasRef.current;
+
+    try {
+      if (canvas) {
+        setIsSaving(true);
+
+        if (canvas.width > targetWidth || canvas.height > targetHeight) {
+          canvas = await scaleDown(canvas, targetWidth);
+        }
+
+        document.body.append(canvas);
+
+        canvas.toBlob(
+          (blob) => {
+            const formData = new FormData();
+            formData.append(props.uploadKey, blob ?? "");
+            formData.append("subject", props.subject);
+            formData.append("uploadKey", props.uploadKey);
+            formData.append("csrf", props.csrfToken);
+
+            if (props.redirect) {
+              formData.append("redirect", props.redirect);
+            }
+
+            if (props.slug) {
+              formData.append("slug", props.slug);
+            }
+
+            fetch(UPLOAD_URL, { method: "POST", body: formData })
+              .then((response) => {
+                if (response.ok) {
+                  return response;
+                } else
+                  throw Error(
+                    `Server returned ${response.status}: ${response.statusText}`
+                  );
+              })
+              .then((response) => {
+                setIsSaving(false);
+                const $modalToggle = document.getElementById(
+                  props.id
+                ) as HTMLInputElement | null;
+                if ($modalToggle) {
+                  $modalToggle.checked = false;
+                }
+                document.location.reload();
+              })
+              .catch((err) => {
+                alert(err);
+              });
+          },
+          IMAGE_MIME,
+          IMAGE_QUALITY
+        );
+      }
```

Comment:
What should be done before the error is thrown?

- f.e. modalToggle.checked = false
- maybe crop and image source reset?